### DISCUSSION TITLE
Allow separate parameters or blocks of them to use different samplers

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1556] [Ideas/Science]
Not sure how easy this is to do in Pints but will likely be very useful for bigger models, with correlated parameters.

**Comment by MichaelClerx** [comment]
~@ben18785 is this a duplicate of #646 ?~

**Comment by MichaelClerx** [comment]
If you could combine this with #646 it would probably make a good project.

I think it'd largely be a feature of the MCMCController (unless you wanted to allow e.g. ABC or nested sampling for some blocks?), so a possible road map would be:

- work out a single ask() call (in a script), that can use multiple MCMC samplers on the same LogPDF. The MCMCSampler class doesn't know about the problem it's optimising (only it's `n_parameters`) so that should be fairly straightforward
- make a call to tell --> now it get's a bit trickier, as some samplers could return `None` here?
- work out what to do in this case
- wrap it up in a loop, see if it works
- depending on how much code it is, add it to `MCMCController`, or write a separate controller
